client/src/main/java/io/pravega/client/batch/impl/SegmentRangeImpl.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.batch.impl;
+
+import com.google.common.annotations.Beta;
+import com.google.common.base.Preconditions;
+import io.pravega.client.batch.SegmentRange;
+import io.pravega.client.segment.impl.Segment;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+
+/**
+ * Implementation of {@link SegmentRange}.
+ */
+@Beta
+@Builder
+@ToString
+public class SegmentRangeImpl implements SegmentRange {
+
+    /**
+     * Segment to which the metadata relates to.
+     */
+    @NonNull
+    @Getter(value = AccessLevel.PACKAGE)
+    private final Segment segment;
+
+    /**
+     * Start offset for the segment.
+     */
+    @Getter(value = AccessLevel.PACKAGE)
+    private final long startOffset;
```

Comment:
Right, the implementation has offsets like I expected, but the interface doesn't, which makes it odd. I can have an implementation of  `SegmentRange` that has no offsets whatsoever.

Reply:
Updated the `SegmentRange` interface to include the following two APIs.
```
long getStartOffset();
long getEndOffset();
```